openapi-scala/src/main/scala/com/enfore/apis/repr/TypeRepr.scala

```diff
@@ -7,12 +7,16 @@ import cats.syntax.functor._
 
 sealed trait TypeRepr {
   val typeName: String
+  def packDefault(value: TypeRepr.PrimitiveValue): TypeRepr
+  def default: Option[_]
+  def defaultString: Option[String] =
+    throw new IllegalAccessError(
+      s"$typeName with type ${this.getClass.getName} has no default value implementation. This should never have happened. Default case encountered is ${default.toString}"
+    )
```

Comment:
why do you supply this throwing implementation? You could just leave it open as for `default` or `packDefault`

Reply:
There are more exceptional cases than normal ones. That's why I thought it was better to have a default implementation than to implement in all places a function that would only raise the same error.